src/agent/transform-headers-agent.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-underscore-dangle */
+const http = require('http');
+const WrappedAgent = require('./wrapped-agent');
+
+const { _storeHeader } = http.OutgoingMessage.prototype;
+
+class TransformHeadersAgent extends WrappedAgent {
+    transformRequest(request) {
+        const headers = {};
+        const hasConnection = request.hasHeader('connection');
+        const hasContentLength = request.hasHeader('content-length');
+        const hasTransferEncoding = request.hasHeader('transfer-encoding');
+        const hasTrailer = request.hasHeader('trailer');
+        const keys = request.getHeaderNames();
+
+        for (const key of keys) {
+            headers[key] = request.getHeader(key);
+            request.removeHeader(key);
+        }
+
+        if (!hasConnection) {
+            const shouldSendKeepAlive = request.shouldKeepAlive && (hasContentLength || request.useChunkedEncodingByDefault || request.agent);
+            if (shouldSendKeepAlive) {
+                headers[this.transformHeader('connection')] = 'keep-alive';
+            } else {
+                headers[this.transformHeader('connection')] = 'close';
+            }
+        }
+
+        if (!hasContentLength && !hasTransferEncoding) {
+            if (!hasTrailer && !request._removedContLen && typeof request._contentLength === 'number') {
+                headers[this.transformHeader('content-length')] = request._contentLength;
+            } else if (!request._removedTE) {
+                headers[this.transformHeader('transfer-encoding')] = 'chunked';
+            }
+        }
+
+        const sorted = Object.keys(headers)/* .sort(this.sort) */;
+
+        for (const key of sorted) {
+            request.setHeader(this.transformHeader(key), headers[key]);
+        }
+    }
+
+    addRequest(request, options) {
+        request._storeHeader = (...args) => {
+            this.transformRequest(request);
+
+            return _storeHeader.call(request, ...args);
+        };
+
+        return super.addRequest(request, options);
+    }
+
+    transformHeader(header) {
+        return header.split('-').map((part) => {
+            return part[0].toUpperCase() + part.slice(1);
+        }).join('-');
+    }
+
+    sort(a, b) {
```

Comment:
Would be nice to add a few lines about how this works and maybe have a name that represents it. E.g. the `sortedHeaders` is pulled out of `this`, but the variable is not defined in this class and neither in the parent class. So where does it come from? How does it sort?

Reply:
It's just a PoC. It needs to be integrated with `header-generator.`